sweep/sweeper.go

```diff
@@ -322,9 +356,17 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch,
 				listeners:        []chan Result{input.resultChan},
 				input:            input.input,
 				minPublishHeight: bestHeight,
+				feeRate:          input.feeRate,
 			}
 			s.pendingInputs[outpoint] = pendInput
 
+			// We'll add the input to a fee rate bucket based on the
+			// fee preference specified by the client so that it can
+			// be batched with the other inputs with similar fee
+			// rates. We can ignore the error as the fee rate has
+			// already been validated.
+			s.bucketManager.addInput(pendInput)
```

Comment:
Is this the best split of responsibilities between `UtxoSweeper` and `bucketManager`? Both keeping sets of inputs looks like it isn't as clear as it could be.

If you are keeping a global set of inputs anyway, you may just as well determine the buckets on the fly when you need them. It would basically be a function that takes a list of (input) fee rates and returns a mapping of those input fee rates to the fee rate to be used for the tx. Clustering the input fee rates identical to how this currently happens in the bucketManager.

Reply:
There's still a single input instance, but multiple references to a given input now exist. I think your second comment is a matter of the degree of encapsulation. As is now, the buckets are a black box to the sweeper until it actually needs to sweep something, but then even from it's PoV they're just another layer above the input lists. Additionally IMO having the new behavior be encapsulated as it is now makes it easier to unit test and also modify exactly _what_ the buckets consist of (for example, confs vs fee rates) in the future. 

I'd also expect the re-clustering logic (when we need to re-target due to a missed confirmation target) to live in this new sub-system as well.

Reply:
I don't see a downside to determining the clusters only when you need them and it allows more dynamic schemes.

If you want to stay with the buckets as they currently are, I'd at least separate concerns more. If a bucket is more or less operating independently, there must be code inside `UtxoSweeper` that can be moved to the bucket. For example creating the txes for a bucket.

Reply:
Done.